core/src/main/scala/org/apache/spark/sql/store/StoreUtils.scala

```diff
@@ -120,7 +121,8 @@ object StoreUtils {
   }
 
   private[sql] def getBucketPreferredLocations(region: PartitionedRegion,
-      bucketId: Int, forWrite: Boolean): Seq[String] = {
+      bucketId: Int, forWrite: Boolean,
+      preferPrimaries: Boolean = false): Seq[String] = {
     if (forWrite) {
```

Comment:
We could have the condition like this no?
if (forWrite || preferPrimaries)

Reply:
forWrite will select only the primary. This change will still have all of primary and secondaries albeit with primary at the start (i.e. in worst case if query partition can't be scheduled on a primary for long time then it will use secondary).

Reply:
yes thought so. Was just thinking will there be a situation like that. Because getNodeForBucketWrite is a blocking call only afaik so either it will get or that call will hang. Isn't it?

Reply:
Not that. Spark will reschedule the partitions if cores are busy on primary after sometime.

Reply:
Ok. Thanks.